src/pages/sign-transaction/sign-transaction.tsx

```diff
@@ -40,23 +41,27 @@ function SignTransactionBase(): JSX.Element | null {
   const [, setFeeRate] = useFeeRateState();
   const feeSchema = useFeeSchema();
 
-  const onSubmit = useCallback(async () => {
-    setIsLoading();
-    await handleBroadcastTransaction();
-    setIsIdle();
-    setFeeEstimations([]);
-    setFee(null);
-    setFeeRate(null);
-    return () => setBroadcastError(null);
-  }, [
-    handleBroadcastTransaction,
-    setBroadcastError,
-    setFee,
-    setFeeEstimations,
-    setFeeRate,
-    setIsIdle,
-    setIsLoading,
-  ]);
+  const onSubmit = useCallback(
+    async values => {
+      setFee(stxToMicroStx(values.txFee).toNumber());
+      setIsLoading();
+      await handleBroadcastTransaction();
+      setIsIdle();
+      setFeeEstimations([]);
+      setFee(null);
+      setFeeRate(null);
```

Comment:
Unrelated to this PR, but I see this as a problem caused by keeping ephemeral form values in global state. We need to manually reset them.

Reply:
I need to think on this bc the fee is used in places like generating the signed tx so it has to be global too, right? So, there is form state which I have as `txFee` and a global fee state as `fee`. Is there a better way to handle this? Thinking.

Reply:
I want to make sure I implement things correctly with Formik so happy to go back and refactor. I haven't used it before.